### DISCUSSION TITLE
Emit Events in a Flow Operation

### DISCUSSION THREAD

**Discussion by erondpowell** [directus/directus#15737] [Archive]
What does "Emit Events" do in the context of the create item, update item and delete item operations in a flow?

When we say "emit events", I'm assuming we are talking about this https://nodejs.org/api/events.html#events

But what effectively happens within the Directus app? I know that "Emit Events" determines whether the event is detected by other flows... But are there any other resources impacted by this? Perhaps in other features of the app or in custom extensions?

(CC @br41nslug @licitdev)

<img width="834" alt="Screen Shot 2022-09-24 at 1 13 07 PM" src="https://user-images.githubusercontent.com/62306630/192112697-b8fc9756-ea37-4665-81bd-0a90508d3fad.png">

**Comment by br41nslug** [comment]
The "Emit Events" option here prevents the "Create Data"/"Update Data" operations from triggering the `items.create`/`items.update` events. This is useful in the situation where you have a flow being triggered by `<collection>.items.update` which then tries to update another (or the same) record in that `<collection>` which would otherwise result in an infinite loop.
